### DISCUSSION TITLE
Exception: No weights file found for element: O

### DISCUSSION THREAD

**Discussion by ZengZezhu** [MarsalekGroup/aml#19] [Q&A] [answered]
Dear AML Team,

I just want to use the aml to generate a NNP potential of water. When I used the script of "create-model.py" prepared in the examples, I encountered an issue like below:

Traceback (most recent call last):
  File "/lustre1/u/u3004964/using_n2p2/aml-committee/water-cheng/create-model.py", line 35, in <module>
    n2p2.train(structures, n_epoch=2)
  File "/lustre1/u/u3004964/software/aml/aml/mlp.py", line 206, in train
    self._read_parameters(directories)
  File "/lustre1/u/u3004964/software/aml/aml/mlp.py", line 160, in _read_parameters
    model_parameters.append(self._read_output_train(directory))
  File "/lustre1/u/u3004964/software/aml/aml/mlp.py", line 449, in _read_output_train
    raise Exception(f'No weights file found for element: {element:s}')
Exception: No weights file found for element: O

I carefully prepared the input files (including input.data, input.nn and scaling,data) for training the NNP by using n2p2, and I also tested and found that it is correct if I directly run 'nnp-train' in the path of 'final-training/train-000'. Could you tell me what should I do for this issue? Many thanks. 

I am also confused about the aml function, does aml can be sued to train the NNP potential? or we need to use the n2p2 to train and then use the aml to generate the C-NNP? 

Cheers,
zezhu

**Comment by ZengZezhu** [comment]
Hi,

Now I solved this issue by a pre-fitting when performing the create-model.py, so I think that aml cannot be sued to fit the NNP. 

But I am still a little confused by the MD simulation of using C-NNP in lammps. I see that the final model just is a simple combination of 8 NNPs located on 8 different folders. Is it possible to use the final C-NNP model to perform MD simulations in lammps? Because I do not find the final C-NNP potential in the model folder, I don't know what should I do to average these 8 NNPs. I see the example of MD simulations in '04-C-NNP-MD' only gives the input of CP2K.:)

**Reply by cschran** [reply]
Committee NNPs are currently only available in `CP2K`, but there is an open `n2p2` [PR](https://github.com/CompPhysVienna/n2p2/pull/92) which implements committees for lammps. This PR can already be used.

**Comment by cschran** [comment, marked as answer]
Hi Zezhu,

thanks for your interest in our work.

`AML` can be faithfully used to train a model with `n2p2`. This is a crucial component of our active learning process.

Internally, `AML` calls `nnp-scaling/train`, but you need to make sure that both are in your path. Furthermore, `mpirun` might cause problems in your particular case. To trouble shoot, try running each training process on a single core (`n_core_task = 1`).

**Reply by OndrejMarsalek** [reply]
Just to further explain, the current version of the code is quite basic in that it does not do proper error checking on the subprocesses (like n2p2 executables) that it launches. That means that if they fail for whatever reason, you only get an error once AML tries to read their output and does not find it. As Christoph says, the most common reason for this failure is the runtime environment - the executable is not found, MPI does not work properly, or something similar. You want to make sure you can run n2p2 "by hand" in the same environment where you're running AML.

The development version of AML that we should be making public soon fixes this and will have better error reporting when subprocesses fail.

**Comment by ZengZezhu** [comment]
Hi,

Thank you for your information! I got it.

Cheers,
zezhu
